src/server/system_services/system_store.js

```diff
@@ -740,9 +740,8 @@ class SystemStore extends EventEmitter {
             });
         });
 
-        const bulk_results = await Promise.all(Object.values(bulk_per_collection).map(
-            bulk => bulk.length && bulk.execute({ j: true })
-        ));
+
+        const bulk_results = await db_client.instance().execute_multiple_bulks(bulk_per_collection);
```

Comment:
rollbacking all collections queries on failures of related collections in a single make_changes() can be a great value for our db validity, 
but it can cause unexpected failures if a single make_changes() updates/inserts/removes non related queries (rollback one query due to failure of non related query)
I would check that every call to make_changes() make sense in that way.

Reply:
@romayalon,
I am not sure what could be not related queries in a single make_changes() function call.
On what basis we can say that two queries are related or not. This could be very complex to handle and could be bug prone if we do not have simple way of segregating queries. 
I have a limited understanding on this so could not visualize the tradeoff. 
@dannyzaken please provide your input as well.

For the refreshed patch, I have not made changes for this comment. Once we decide what to do next, I will send  modified code.